docs/dev/rfc/odfe-cli-rest-api-as-commands.md

```diff
@@ -0,0 +1,298 @@
+# RFC: Support REST api as command
+
+## Objective
+
+This document describes how to support [rest api](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html) in a generic way.
+As of now, elasticsearch support api in different categories like index api, cat api, cluster api, document api, etc to
+configure and access their features. It will take tremendous amount of time and effort to provide every api as native commands.
+Hence, we will be categorizing those REST API based on GET/PUT/POST/DELETE  and allow users to perform their request 
+in a generic way. This will support any future API without any additional support. It also benefits plugin owners while on-boarding their plugin. 
+
+## Available commands
+
+1. [get command](./odfe-cli-rest-api-as-commands.md#1-get-command)
+2. [post command](./odfe-cli-rest-api-as-commands.md#2-post-command)
+3. [put command](./odfe-cli-rest-api-as-commands.md#3-put-command)
+4. [delete command](./odfe-cli-rest-api-as-commands.md#4-delete-command)
+
+## Common parameters
+
+The following options can be applied to all of the odfe-cli commands. These parameters are supported by server as mentioned [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html).
+In future  more parameters or include multiple values will be supported that can be added as flag to any commands.
+
+
+
+|Parameter name|Short hand	|Default|Usage|
+|---------------|-----------|-------|-------|
+|--pretty	    |-b	        | false	|if true, the response returned will be pretty formatted	|
+|--output-format |-o	|json	|other value is yaml. The result to be returned in the (if supported) more readable yaml format.	|
+|--human	| N/A	| false	|the output statistics are returned in a human friendly format. (e.g. `"exists_time": "1h"` or `"size": "1kb"`) instead of (e.g. `"exists_time_in_millis": 3600000` or `"size_in_bytes": 1024`). 	|
+|--filter-path	|-f	| N/A	|filter the response fields returned by cluster. It accepts a comma separated list of filters expressed with the dot notation.  eg: `--filter-path=“took,hits.hits._id,hits.hits._score”`	|
+
+
+
+## 1. get command
```

Comment:
Is it necessary to enumerate these? Im not sure how I feel on it.

Reply:
So you don't want a numbered list?

Reply:
On second thought, I think it is okay.